### PR TITLE
Update dependencies, switch to eslint

### PR DESCRIPTION
**Pull request by QWp6t**
Note: I'm targeting sage-9 branch. This isn't going to master. J

Using the latest versions of npm and node.js, I was able to update all of the npm dependencies to their latest stable versions. I then tested by running gulp. Everything passes, as I'm sure travis will confirm.

I switched to eslint because I'm pretty sure we all agreed that it was the right move, but also because jshint has a history of playing fast and loose with semver, which makes it difficult to have it as a dependency. Since eslint can do everything jshint can do (except validate json files, apparently -- note that bower.json is no longer validated in `gulp lint` task) and it also does code style, I figured it was a better choice.

I notice that travis failed. I'm not sure how to get travis on a newer version of npm (or node.js?), but you can test this branch yourself; everything should pass.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,24 +1,37 @@
...
+    - g++-4.8
 
 cache:
   directories:
     - bower_components
     - node_modules
+    - vendor
 
 install:
+  - rm -rf ~/.nvm && git clone https://github.com/creationix/nvm.git ~/.nvm && (cd ~/.nvm && git checkout `git describe --abbrev=0 --tags`) && source ~/.nvm/nvm.sh && nvm install $TRAVIS_NODE_VERSION
```
Comment (swalkinshaw) [inline .travis.yml]: Makes a little more logical sense to put `rm -rf ~/.nvm` in `before_install`.

Comment (swalkinshaw) [inline .travis.yml]: is this needed because we're using PHP as the main language? Because Travis does support nvm natively.

Comment (swalkinshaw) [inline .travis.yml]: `nvm` does already exist though if that makes it simpler: http://entulho.fiatjaf.alhur.es/guias/how-to-use-node-along-with-other-language-on-travis-ci/

Comment (QWp6t) [inline .travis.yml]: I'm assuming it's needed. I'm not sure, though; I didn't test without it. I did see that Travis supports nvm natively, but I assume that's only for node_js containers. 

I poached this from some random nerd online: http://austinpray.com/ops/2015/09/20/change-travis-node-version.html

**Comment by swalkinshaw** [comment]
:+1: for eslint

**Comment by ntwb** [comment]
Agree with eslint also :+1: 

Noting this PR will also close https://github.com/roots/sage/issues/1361

**Comment by austinpray** [comment]
Eslint 👌🏻

Did you make sure the jscs rules don't conflict with the eslint recommended rules? Do we need to be using both?

**Pull request merged**